### DISCUSSION TITLE
Deserialize as primitive components?

### DISCUSSION THREAD

**Discussion by javafanboy** [activej/activej#322] [Q&A]
I am working on a project where I send large numbers of serialized messages from multiple clients to an aggregator for processing. In the clients the data is in the format of Java POJOs but in the aggregator I instead wants to use some pre-allocated arrays to more efficiently and with less GC process this data. Preferably I would like to de-serialize without re-creating the serialized objects but rather directly read the primitive data content of each encoded object and add to my arrays without any object allocation...
What would be the simplest way to serialize my data using ActiveJ Serialization to be able to do this?

**Comment by eduard-vasinskyi** [comment]
Hi, @javafanboy

That depends on how you created a serializer (`BinarySerializer`) in the first place. If it was created manually, then simply reverse the process when deserializing. 

If `BinarySerializer` was instead created via `SerializerFactory` and you are not sure how data was actually serialized, then you can obtain a class file of that BinarySerializer and reverse engineer the deserialization process from there. 

For that, you would need to pass a `DefiningClassLoader` into the `SerializerFactory#create` method when creating a `BinarySerializer`. You need to enable `DefinigClassLoader.Builder#withDebugOutputDir` and set a directory where classes will be generated. After that, you can open that class file and copy the deserialization routine from the `decode()` method.
